plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryConfig.java

```diff
@@ -266,6 +270,25 @@ public BigQueryConfig setQueryResultsCacheEnabled(boolean queryResultsCacheEnabl
         return this;
     }
 
+    public String getQueryLabelFormat()
+    {
+        return queryLabelFormat;
+    }
+
+    @AssertTrue(message = "Incorrect bigquery.job.label-format it may consist of only letters, digits, underscores, commas, spaces, equal signs and predefined values")
+    boolean isFormatValid()
+    {
+        return hasValidPlaceholders(queryLabelFormat, SessionInterpolatedValues.values());
+    }
+
+    @Config("bigquery.job.label-format")
+    @ConfigDescription("Adds `trino_query` label to the BigQuery job with provided value format")
+    public BigQueryConfig setQueryLabelFormat(String queryLabelFormat)
```

Comment:
while it's useful to use common format across connectors for BigQuery it's probably less powerful in the way we use it since all information gets stuffed into single label in BigQuery - I feel this is prematurely adding a feature which would make it harder to evolve over time as people request changes.

What do you think @kokosing @ebyhr - should we not add it at the moment and just label the query id for now?

Alternatively we can convert the config into a boolean toggle and if enabled add one label per each pre-defined placeholder.